core/crypto/crypto_store/key_type.hpp

```diff
@@ -16,52 +16,99 @@ namespace kagome::crypto {
     UNSUPPORTED_KEY_TYPE_ID,
   };
 
-  /**
-   * @brief Key type identifier
-   */
+  /// Key type identifier
   using KeyTypeId = uint32_t;
 
   /**
-   * Types are 32bit integers, which represent encoded 4-char strings
+   * Makes 32bit integer KeyTypeId which represent encoded 4-char strings
    * Little-endian byte order is used
    */
-  enum KnownKeyTypeId : KeyTypeId {
-    // clang-format off
-    KEY_TYPE_BABE = 0x65626162u, // BABE, sr25519
-    KEY_TYPE_GRAN = 0x6e617267u, // GRANDPA, ed25519
-    KEY_TYPE_ACCO = 0x6f636361u, // Account control [sr25519, ed25519, secp256k1]
-    KEY_TYPE_IMON = 0x6e6f6d69u, // I'm Online, sr25519
-    KEY_TYPE_AUDI = 0x69647561u, // Account discovery [sr25519, ed25519, secp256k1]
-    KEY_TYPE_ASGN = 0x6e677361u, // ASGN
-    KEY_TYPE_PARA = 0x61726170u, // PARA
-    KEY_TYPE_BEEF = 0x66656562u, // Beefy, secp256k1
-    // clang-format on
+  constexpr KeyTypeId operator""_key(const char *s, std::size_t size) {
+    return (static_cast<KeyTypeId>(s[0]) << (CHAR_BIT * 0))
+         | (static_cast<KeyTypeId>(s[1]) << (CHAR_BIT * 1))
+         | (static_cast<KeyTypeId>(s[2]) << (CHAR_BIT * 2))
+         | (static_cast<KeyTypeId>(s[3]) << (CHAR_BIT * 3));
+  }
+
+  class KeyType {
```

Comment:
```
constexpr uint32_t operator""_key(...);

enum class KeyType : uint32_t {
  BABE = "babe"_key,
```

May forward declare enum for return type, but need to cast it back when initializing enum (compile error).
```
enum class KeyType : uint32_t;
constexpr KeyType operator""_key(...);
enum class KeyType : uint32_t {
  BABE = (uint32_t)"babe"_key,
```